Mathlib/LinearAlgebra/CliffordAlgebra/Grading.lean

```diff
@@ -165,33 +176,59 @@ theorem evenOdd_induction (n : ZMod 2) {P : ∀ x, x ∈ evenOdd Q n → Prop}
           P (ι Q m₁ * ι Q m₂ * x)
             (zero_add n ▸ SetLike.mul_mem_graded (ι_mul_ι_mem_evenOdd_zero Q m₁ m₂) hx))
     (x : CliffordAlgebra Q) (hx : x ∈ evenOdd Q n) : P x hx := by
-  apply Submodule.iSup_induction' _ _ (hr 0 (Submodule.zero_mem _)) @hadd
+  apply Submodule.iSup_induction' (C := P) _ (hr 0 (Submodule.zero_mem _)) @hadd
   refine' Subtype.rec _
-  simp_rw [Subtype.coe_mk, ZMod.nat_coe_zmod_eq_iff, add_comm n.val]
+  -- porting note: was `simp_rw [Subtype.coe_mk, ZMod.nat_coe_zmod_eq_iff, add_comm n.val]`
+  -- leanprover/lean4#1926 is to blame.
+  dsimp only [Subtype.coe_mk]
+  let hlean1926 : ∀ val : ℕ, ↑val = n ↔ ∃ k, val = 2 * k + ZMod.val n := by
+    intro val
+    simp_rw [ZMod.nat_coe_zmod_eq_iff, add_comm n.val]
+  have := fun val : ℕ => forall_prop_congr'
+    (q := fun property => ∀ x (hx : x ∈ LinearMap.range (ι Q) ^ val),
+      P x (Submodule.mem_iSup_of_mem { val := val, property := property } hx))
+    (hq := fun property => Iff.rfl) (hp := hlean1926 val)
+  simp_rw [this]; clear this
+  -- end porting note
   rintro n' ⟨k, rfl⟩ xv
-  simp_rw [pow_add, pow_mul]
-  refine' Submodule.mul_induction_on' _ _
-  · intro a ha b hb
-    refine' Submodule.pow_induction_on_left' ((ι Q).range ^ 2) _ _ _ ha
-    · intro r
+  -- porting note: was `simp_rw [pow_add, pow_mul]`
+  -- leanprover/lean4#1926 is to blame.
+  refine (forall_prop_congr' (hq := fun property => Iff.rfl) (
+    show xv ∈ LinearMap.range (ι Q) ^ (2 * k + ZMod.val n)
+        ↔ xv ∈ (LinearMap.range (ι Q) ^ 2) ^ k * LinearMap.range (ι Q) ^ ZMod.val n by
+      simp_rw [pow_add, pow_mul])).mpr ?_
+  -- end porting note
+  intro hxv
+  induction hxv using Submodule.mul_induction_on' with
+  | hm a ha b hb =>
+    induction ha using Submodule.pow_induction_on_left' with
+    | hr r =>
       simp_rw [← Algebra.smul_def]
       exact hr _ (Submodule.smul_mem _ _ hb)
-    · intro x y n hx hy
+    | hadd x y n hx hy ihx ihy =>
       simp_rw [add_mul]
-      apply hadd
-    · intro x hx n y hy ihy
+      apply hadd ihx ihy
+    | hmul x hx n'' y hy ihy =>
       revert hx
-      simp_rw [pow_two]
-      refine' Submodule.mul_induction_on' _ _
-      · simp_rw [LinearMap.mem_range]
-        rintro _ ⟨m₁, rfl⟩ _ ⟨m₂, rfl⟩
+      -- porting note: was `simp_rw [pow_two]`
+      -- leanprover/lean4#1926 is to blame.
+      let hlean1926'' : x ∈ LinearMap.range (ι Q) ^ 2
+          ↔ x ∈ LinearMap.range (ι Q) * LinearMap.range (ι Q) := by
+        rw [pow_two]
```

Comment:
Interestingly the proof fails if I use a `have` instead of a `let`, as then the induction principle is wrong.